src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -82,73 +82,81 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
+     * Then, for each {@code c_a, a <= N}, there must be a corresponding distinct
+     * {@code e_b} such that the {@link MemoryLayout#name() name} of {@code c_a}
+     * {@link Object#equals(Object) equals} the {@link RecordComponent#getName() name} of {@code e_b} and:
      * <ul>
      *    <li>
-     *        <h4>If <em>E<sub>b</sub></em> is a {@link ValueLayout };</h4>
-     *        then <em>C<sub>a</sub></em> must be of the exact type of <em>E<sub>b</sub>'s</em> {@link ValueLayout#carrier() carrier()}<br>
-     *        whereby C<sub>a</sub> = f<sub>a</sub>(MemorySegment ms) = ms.get(E<sub>b</sub>, offset) at the appropriate offset.<br>
+     *        <h4>If {@code e_b} is a {@link ValueLayout };</h4>
+     *        then {@code c_a} must be of the exact type of {@code e_b}'s {@link ValueLayout#carrier() carrier()}<br>
+     *        whereby {@code c_a = f_a(MemorySegment ms) = ms.get(e_b, offset + offsetBetween(layout, e_b))}.<br>
      *    </li>
      *    <li>
-     *        <h4>If <em>E<sub>b</sub></em> is a {@link GroupLayout };</h4>
-     *        then <em>C<sub>a</sub></em> must be of another {@link Record} type <em>R2</em>
-     *        (such that <em>R2</em> {@code !=} <em>R</em>) that can be mapped to <em>E<sub>b</sub></em> via
-     *        a resulting mapper <em>M2</em> =
-     *        {@link #recordMapper(Class) E<sub>b</sub>.recordMapper(C<sub>a</sub>.type())}<br>
-     *        whereby <em>C<sub>a</sub></em> = f<sub>a</sub>(MemorySegment ms) = <em>M2</em>.apply(ms) recursively
-     *        at the appropriate offset.<br>
+     *        <h4>If {@code e_b} is a {@link GroupLayout };</h4>
+     *        then {@code e_a} must be of another {@link Record} type {@code R2}
+     *        (such that {@code R2 != R}) that can be mapped to {@code e_b} via
+     *        another resulting mapper {@code M2 = M(R2, e_b, offset + offsetBetween(layout, e_b))} recursively
+     *        whereby {@code c_a} = {@code f_a(MemorySegment ms) = M2.apply(ms)}<br>
      *    </li>
      *    <li>
-     *        <h4>If <em>E<sub>b</sub></em> is a {@link SequenceLayout };</h4>
-     *        then <em>C<sub>a</sub></em> must be an array <em>C[]<sup>D</sup></em> (of depth <em>D</em>
-     *        and with an array component type <em>C</em>) that can be mapped to <em>E<sub>b</sub></em> via a resulting
-     *        "array mapper" <em>A2</em> obtained via recursively pealing off nested sequence layouts in <em>E<sub>b</sub></em>
-     *        and then (after <em>D</em> pealing operations)
-     *        finally determining the leaf element layout <em>LL</em> = {@link SequenceLayout#elementLayout() elementLayout()}
-     *        and subsequently obtaining a leaf record mapper:
+     *        <h4>If {@code e_b} is a {@link SequenceLayout };</h4>
+     *        then {@code c_a} must be an array {@code C[]^D} (an array of depth {@code D}
+     *        and with an array component type {@code C}) that can be mapped to {@code e_b} via a resulting
```

Comment:
Can't all this now be expressed more simply using recursion? E.g. given the element layout E and the component type C (which could be another array), we compute, for each element in the sequence layout, the mapping between E, C and the corresponding element offset. Then we return new C[] with all the elements so computed. If you propagate the offsets correctly it should "just work" ?